From d343a4d86ac883dc82ef8b7178931d83ab6dffe4 Mon Sep 17 00:00:00 2001
From: Xing Zheng <zhengxing@rock-chips.com>
Date: Tue, 15 May 2018 15:44:06 +0800
Subject: [PATCH] ASoC: rockchip: multicodecs: using stream not clk_id

We need to using stream to know whether is playback
or capture in sys_clk.

Change-Id: If392c0bc6a6b7d9309ca8ee5b33945d42193792e
Signed-off-by: Xing Zheng <zhengxing@rock-chips.com>
---
 sound/soc/rockchip/rockchip_multicodecs.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sound/soc/rockchip/rockchip_multicodecs.c b/sound/soc/rockchip/rockchip_multicodecs.c
index d6abf0c9a7c6..94fc2f25cbe6 100644
--- a/sound/soc/rockchip/rockchip_multicodecs.c
+++ b/sound/soc/rockchip/rockchip_multicodecs.c
@@ -54,13 +54,15 @@ static int rk_multicodecs_hw_params(struct snd_pcm_substream *substream,
 
 	mclk = params_rate(params) * mc_data->mclk_fs;
 
-	ret = snd_soc_dai_set_sysclk(codec_dai, 0, mclk, SND_SOC_CLOCK_IN);
+	ret = snd_soc_dai_set_sysclk(codec_dai, substream->stream, mclk,
+				     SND_SOC_CLOCK_IN);
 	if (ret && ret != -ENOTSUPP) {
 		pr_err("Set codec_dai sysclk failed: %d\n", ret);
 		goto out;
 	}
 
-	ret = snd_soc_dai_set_sysclk(cpu_dai, 0, mclk, SND_SOC_CLOCK_OUT);
+	ret = snd_soc_dai_set_sysclk(cpu_dai, substream->stream, mclk,
+				     SND_SOC_CLOCK_OUT);
 	if (ret && ret != -ENOTSUPP) {
 		pr_err("Set cpu_dai sysclk failed: %d\n", ret);
 		goto out;
-- 
2.35.3

